internal/pkg/runtime/launcher/oci/launcher_linux.go

```diff
@@ -397,9 +407,40 @@ func (l *Launcher) finalizeSpec(ctx context.Context, b ocibundle.Bundle, spec *s
 		return err
 	}
 
+	l.handleVarTmpToTmpSymlink(spec)
+
 	return b.Update(ctx, spec)
 }
 
+func (l *Launcher) handleVarTmpToTmpSymlink(spec *specs.Spec) {
+	tmpResolved := fs.EvalRelative(tmpPath, spec.Root.Path)
+	vartmpResolved := fs.EvalRelative(vartmpPath, spec.Root.Path)
+	sylog.Debugf("Container /tmp resolves to %q", tmpResolved)
+	sylog.Debugf("Container /var/tmp resolves to %q", vartmpResolved)
+
+	// If /var/tmp doesn't point to the same place as /tmp after resolving
+	// symlinks, then there's nothing more to do here.
+	if vartmpResolved != tmpResolved {
+		return
+	}
+
+	// Get the default mounts whose destination path is /var/tmp
+	vartmpDefaultMounts := lo.Filter(l.defaultTmpMounts, func(m specs.Mount, _ int) bool {
+		return (m.Destination == vartmpPath)
+	})
+
+	// Remove default mounts whose destination is /var/tmp from the OCI spec
+	// we're generating.
+	for _, defMount := range vartmpDefaultMounts {
+		spec.Mounts = lo.Filter(spec.Mounts, func(m specs.Mount, _ int) bool {
+			// The first disjunct here is superfluous, logically speaking; it's
+			// here to ensure comparison is as fast as possible. It only makes
+			// sense to bother with DeepEqual() if we know the path matches.
+			return ((m.Destination != vartmpPath) || !reflect.DeepEqual(m, defMount))
```

Comment:
I'm not sure whether `reflect.DeepEqual` is the best thing to filter with here.

In theory, could someone specify a `-B` bind of a workdir var_tmp that would end up with the same struct content as a default bind from a workdir var_tmp?

Also... if we extend our Docker compatible `--mount` syntax to support things other than binds, then it'd be possible for someone to request a tmpfs mount via `--mount`.... which perhaps would be indistinguishable from the 

Can we filter based on the address of the default mount spec.Mount structs... rather than their content, to avoid any issue?

Reply:
Yeah, good point!